src/NexusMods.Abstractions.Loadouts.Synchronizers/IGameLocationsService.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Frozen;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Hashing.xxHash3;
+using NexusMods.Paths;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+[PublicAPI]
+public record struct IndexFileResult(
+    GamePath Path,
+    Hash Hash,
+    Size Size,
+    DateTimeOffset LastModified
+);
+
+[PublicAPI]
+public record IndexGameResult(
+    FrozenDictionary<GamePath, IndexFileResult> NewFiles,
+    FrozenDictionary<GamePath, IndexFileResult> ModifiedFiles,
+    FrozenSet<GamePath> RemovedFiles
+);
+
+[PublicAPI]
+public interface IGameLocationsService
```

Comment:
We should add good documentation to these new interfaces as we go along, to ensure that the scope is well defined and they don't something they are not meant to be.